source/javascripts/components/unified-editor/WorkflowConfig/WorkflowConfigPanel.tsx

```diff
@@ -1,19 +1,36 @@
 import { TabPanel, TabPanels, Tabs } from '@bitrise/bitkit';
+import { useFormContext } from 'react-hook-form';
+import { useShallow } from 'zustand/react/shallow';
+import useSearchParams from '@/hooks/useSearchParams';
+import useBitriseYmlStore from '@/hooks/useBitriseYmlStore';
 import WorkflowConfigHeader from './components/WorkflowConfigHeader';
 import ConfigurationTab from './tabs/ConfigurationTab';
 import PropertiesTab from './tabs/PropertiesTab';
 import WorkflowConfigProvider from './WorkflowConfig.context';
-import { WorkflowConfigTab } from './WorkflowConfig.types';
-
-type Props = {
-  workflowId: string;
-};
+import { FormValues, WorkflowConfigTab } from './WorkflowConfig.types';
+import useRenameWorkflow from './hooks/useRenameWorkflow';
 
 const WorkflowConfigPanelContent = () => {
+  const form = useFormContext<FormValues>();
+  const originalName = form.formState.defaultValues?.properties?.name ?? '';
+  const [, setSearchParams] = useSearchParams();
+  const updateWorkflow = useBitriseYmlStore(useShallow((s) => s.updateWorkflow));
+  const renameWorkflow = useRenameWorkflow((newWorkflowId) => {
+    setSearchParams((searchParams) => ({
+      ...searchParams,
+      workflow_id: newWorkflowId,
+    }));
+  });
+
+  const handleChange = form.handleSubmit(({ properties: { name, ...properties } }) => {
+    updateWorkflow(originalName, properties);
+    renameWorkflow(name);
```

Comment:
If we get rid of the `useReameWorkflow`
```suggestion
    renameWorkflow(originalName, name);
```

Reply:
Why?